.github/workflows/ci.yml

```diff
@@ -270,7 +271,7 @@ jobs:
           IMAGE: ${{ steps.build-docker-image.outputs.image }}
         shell: bash
       - name: Test Image
-        if: steps.prepare-meta.outputs.has-image && (github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/trying')
+        if: (steps.prepare-meta.outputs.has-image && (github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/trying')) && matrix.target != 'cross'
```

Comment:
I think this is the right approach, but feedback would be nice. I want separate logic for our cross image and everything else, so I've split the tests into 2: one for cross and one for everything else.